test/test_sum.jl

```diff
@@ -0,0 +1,15 @@
+srand(123)
+
+f1 = NormL2()
+f2 = NormL2()
+f = Sum(f1, f2)
+xtest = randn(10)
+
+result = f1(xtest) + f2(xtest)
+@test f(xtest) ≈ result
+
+grad1, val1 = gradient(f1, xtest)
+grad2, val2 = gradient(f2, xtest)
+
+gradsum, valsum = gradient(f, xtest)
```

Comment:
Here I would use `gradient!`, passing as result vector `gradsum` which is initialized to something nonzero (to test the fix in my comment above).